x/poe/module.go

```diff
@@ -190,7 +192,7 @@ func (am AppModule) InitGenesis(ctx sdk.Context, cdc codec.JSONCodec, data json.
 	switch diff, err := contract.CallEndBlockWithValidatorUpdate(ctx, addr, am.twasmKeeper); {
 	case err != nil:
 		panic(fmt.Sprintf("poe sudo call: %s", err))
-	case len(diff) == 0:
+	case seedMode && len(diff) == 0:
```

Comment:
Nice catch. On non-seed mode, there may be existing validators and no diff.

What really needs to be checked however, is if there are any validators after the diff is applied. Maybe there is a better way to check that?

Reply:
Shouldn't tendermint check that? I was assuming some error when no validators are left

Reply:
Found this in cosmos-sdk: https://github.com/cosmos/cosmos-sdk/blob/v0.45.4/baseapp/abci.go#L67-L72

Reply:
It was actually required to return the list of active validators. I am also checking for non empty list.